### PR TITLE
`k-input-validator`

### PR DESCRIPTION
**Pull request by distantnative**
### Enhancements
- Added `k-input-validator` helper element to provide native form validation for complex/group inputs and fields

### For review team

- [x] Add lab and/or sandbox examples (wherever helpful)
- [x] Add changes & docs to release notes draft in Notion

### CODE REVIEW

File: panel/lab/components/inputs/list/index.vue [hunk 1]
```diff
@@ -33,6 +33,14 @@
...
 				<k-list-input :disabled="true" :value="value" @input="value = $event" />
 			</k-lab-example>
+
+			<k-lab-example label="Minlength: 5">
+				<k-list-input :minlength="5" :value="value" @input="value = $event" />
+			</k-lab-example>
+
+			<k-lab-example label="Maxlength: 10">
+				<k-list-input :maxlength="10" :value="value" @input="value = $event" />
+			</k-lab-example>
```
Comment (bastianallgeier) [inline panel/lab/components/inputs/list/index.vue]: They both don't really seem to do anything.

**Review by bastianallgeier — COMMENTED** [review]
I really like the idea of the input validator. I think we need to add an option though to provide an element that the validity check can focus on. Right now, there are still console issues because of it and no message can be displayed on submit.

**Comment by distantnative** [comment]
> provide an element that the validity check can focus on

You mean a normal, non-input element to be focussed?

**Comment by bastianallgeier** [comment]
@distantnative I haven't tried if that actually works. But e.g. for the checkboxes we could focus the first checkbox at least. I'm not sure how we should properly handle it when there's actually no valid form element to focus on – e.g. for blocks.

**Comment by distantnative** [comment]
@bastianallgeier I fear we cannot target any other element that isn't a descendant of `k-input-validator`:

```
Failed to execute 'setValidity' on 'ElementInternals': The Element argument should be a shadow-including descendant of the target element.
```

So we would have to wrap the whole component in it... which makes it a little nasty/ugly.

However, according to the docs, any HTMLElement can be the anchor then.

**Comment by distantnative** [comment]
But this seems to be fake news. HTMLElement as anchor still causes console error for me.

**Comment by distantnative** [comment]
@bastianallgeier I think I found a way

**Review by bastianallgeier — APPROVED** [review]
Awesome. It works really well like this 💛

**Comment by bastianallgeier** [comment]
Could you take care of the conflicts after the class attributes? I think I just broke something while trying for the checkboxes input.

**Review by bastianallgeier — APPROVED** [review]

**Pull request merged**